src/components/Planning/ProjectProgrammedDialog/ProjectProgrammedSearch.tsx

```diff
@@ -0,0 +1,168 @@
+import { getProjectsWithParams } from '@/services/projectServices';
+import { arrayHasValue } from '@/utils/common';
+import { SearchInput } from 'hds-react/components/SearchInput';
+import { FC, memo, useCallback, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useOptions } from '@/hooks/useOptions';
+import { IProgrammedProjectSuggestions, ISearchRequest } from '@/interfaces/searchInterfaces';
+import { IClass } from '@/interfaces/classInterfaces';
+import { useAppSelector } from '@/hooks/common';
+import {
+  selectAllClasses,
+  selectSelectedClass,
+  selectSelectedSubClass,
+} from '@/reducers/classSlice';
+import SelectedProjectCard from './SelectedProjectCard';
+
+interface ISearchState {
+  searchWord: string;
+  searchedProjects: Array<IProgrammedProjectSuggestions>;
+}
+interface IProjectSearchProps {
+  onProjectClick: (value: IProgrammedProjectSuggestions | undefined) => void;
+  projectsForSubmit: Array<IProgrammedProjectSuggestions>;
+  onProjectSelectionDelete: (projectName: string) => void;
+}
+
+const buildBreadCrumbs = (path: string, classes: Array<IClass>): Array<string> =>
+  path.split('/').map((p) => classes.find((c) => c.id === p)?.name || '');
```

Comment:
can a bread crumb list contain an empty string as element?

Reply:
It cannot, but removing the empty string will give type error in case classes.find cannot find anything